cumulus/polkadot-parachain/polkadot-parachain-lib/src/fake_runtime_api/mod.rs

```diff
@@ -17,5 +17,50 @@
 //! In an ideal world this would be one runtime which would simplify the code massively.
 //! This is not an ideal world - Polkadot Asset Hub has a different key type.
 
-pub mod asset_hub_polkadot_aura;
-pub mod aura;
+mod utils;
+
+use crate::common::types::CustomBlock;
+use utils::{impl_node_runtime_apis, imports::*};
+
+pub mod asset_hub_polkadot {
+	use super::{utils::imports::*, *};
+	use parachains_common::AssetHubPolkadotAuraId;
+
+	type Block = CustomBlock<u32>;
+	pub struct FakeRuntime;
+	impl_node_runtime_apis!(FakeRuntime, Block, AssetHubPolkadotAuraId);
+}
+
+pub mod u32_block {
+	use super::*;
+
+	type Block = CustomBlock<u32>;
+	pub mod aura_sr25519 {
+		use super::*;
+		struct FakeRuntime;
+		impl_node_runtime_apis!(FakeRuntime, Block, sp_consensus_aura::sr25519::AuthorityId);
+	}
+
+	pub mod aura_ed25519 {
+		use super::*;
+		struct FakeRuntime;
+		impl_node_runtime_apis!(FakeRuntime, Block, sp_consensus_aura::ed25519::AuthorityId);
+	}
+}
+
+pub mod u64_block {
```

Comment:
Damn, this is unfortunate. Similar to the above, it _might_, depending on what is being made parameterize-able, explore exponentially. 

Although, not everything will affect this, right? 

 I wish we could pull a more blanket trick here and make the `node <-> runtime` communication not be _not-type-safe_. This degree of type-safety doesn't make sense for a node that is meant to work with various runtimes, and afaik is mainly the legacy of native runtime days.